packages/contracts/contracts/TestContracts/invariants/echidna/EchidnaBraindeadFeedTester.sol

```diff
@@ -0,0 +1,310 @@
+import "../../../Dependencies/AuthNoOwner.sol";
+import {PriceFeedOracleTester} from "../../PriceFeedOracleTester.sol";
+import {MockAlwaysTrueAuthority} from "../../MockAlwaysTrueAuthority.sol";
+
+import "../PropertiesDescriptions.sol";
+
+import "@crytic/properties/contracts/util/Hevm.sol";
+
+contract EchidnaBraindeadFeedTester is
+    PropertiesConstants,
+    PropertiesAsserts,
+    PropertiesDescriptions
+{
+    BraindeadFeed internal braindeadFeed;
+    PriceFeedOracleTester internal primaryTester;
+    PriceFeedOracleTester internal secondaryTester;
+    PriceFeed internal priceFeed;
+    MockAggregator internal collEthCLFeed;
+    MockAggregator internal ethBtcCLFeed;
+    MockAlwaysTrueAuthority internal authority;
+    MockFallbackCaller internal fallbackCaller;
+
+    uint256 internal constant MAX_PRICE_CHANGE = 5e18;
+    uint256 internal constant MAX_ROUND_ID_CHANGE = 5;
+    uint256 internal constant MAX_UPDATE_TIME_CHANGE = 2 days;
+    uint256 internal constant MAX_STATUS_HISTORY_OPERATIONS = 32;
+    uint256 internal constant MAX_REVERT_PERCENTAGE = 0.1e18;
+
+    // NOTE: These values imply BIAS, you should EDIT THESE based on the target application
+    uint256 internal constant MAX_FALLBACK_VALUE = type(uint128).max;
+    uint256 internal constant MIN_FALLBACK_VALUE = 0; // NOTE: 0 is important as it signals an error / broken to the price feed
+
+    // https://etherscan.io/address/0x86392dc19c0b719886221c78ab11eb8cf5c52812#readContract
+    // Aggregator is here: https://etherscan.io/address/0x716BB759A5f6faCdfF91F0AfB613133d510e1573#readContract
+    // Max and Min are from the aggregator
+    uint256 internal constant MAX_ETH_VALUE = 100000000000000000000;
+    uint256 internal constant MIN_ETH_VALUE = 1000000000000000;
+
+    // https://etherscan.io/address/0xac559f25b1619171cbc396a50854a3240b6a4e99#code
+    // https://etherscan.io/address/0x0f00392FcB466c0E4E4310d81b941e07B4d5a079
+    uint256 internal constant MAX_BTC_VALUE = 1000000000;
+    uint256 internal constant MIN_BTC_VALUE = 10000;
+
+    uint256 internal statusHistoryOperations = 0;
+    IPriceFeed.Status[MAX_STATUS_HISTORY_OPERATIONS] internal statusHistory;
+
+    constructor() payable {
+        authority = new MockAlwaysTrueAuthority();
+        collEthCLFeed = new MockAggregator();
+        ethBtcCLFeed = new MockAggregator();
+
+        hevm.roll(123123131);
+
+        collEthCLFeed.setLatestRoundId(2);
+        collEthCLFeed.setPrevRoundId(1);
+        collEthCLFeed.setUpdateTime(block.timestamp);
+        collEthCLFeed.setPrevUpdateTime(block.timestamp - 1);
+        collEthCLFeed.setPrice(1 ether - 3);
+        collEthCLFeed.setPrevPrice(1 ether - 1337);
+
+        ethBtcCLFeed.setLatestRoundId(2);
+        ethBtcCLFeed.setPrevRoundId(1);
+        ethBtcCLFeed.setUpdateTime(block.timestamp);
+        ethBtcCLFeed.setPrevUpdateTime(block.timestamp - 1);
+        ethBtcCLFeed.setPrice(3 ether - 2);
+        ethBtcCLFeed.setPrevPrice(3 ether - 42);
```

Comment:
I know roundId 0 has caused issues to projects in the past @GalloDaSballo do you remember anything about that here?

are there any other values to call out here or are they just nice valid initial values?

Reply:
We need to roll to ensure we don't overflow

0 would cause revert to PriceFeed due to underflow
This should be caught by the tinfoilcall